### PR TITLE
New package: CONZEInformatik.Vistrax

### PR DESCRIPTION
**Pull request by FaserF**
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/123504)

### CODE REVIEW

**Comment by stephengillie** [comment]
Hi @FaserF,

This package had an error during Manual Validation install:
image

I wasn't able to find which Windows Feature is being modified. Do you know?

**Comment by FaserF** [comment]
Unfortunatly I am not sure what windows feature is being modified. The only requirement I have found is "Microsoft .NET Framework 4.6.2 or later". On my Windows 11 machine I am unable to reproduce that issue. 

While searching the web for that issue, I always find something that this issue is indeed related to a missing .NET version.

File: manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+InstallerType: zip
+Scope: machine
+InstallModes:
+- interactive
+- silent
+InstallerSwitches:
+  Silent: /quiet
+  SilentWithProgress: /passive
+UpgradeBehavior: install
+ReleaseDate: 2023-03-02
```
Comment (stephengillie) [inline manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:16]:
```suggestion
ReleaseDate: 2023-03-02
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.DotNet.Framework.DeveloperPack_4
```

Comment (FaserF) [reply manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:16]: Thank you!
The 64bit one dependency should be fine from what I have read on the manufacturer website.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 4:25:42 PM:  > 2023-10-27 16:25:41.208 [CLI ] ShellExecute installer failed: 3010 

`3010` often means `success but reboot needed` - is that the case here? We could add it under `InstallerSuccessCodes`.

File: manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Dependencies:
+  PackageDependencies:
+    - PackageIdentifier: Microsoft.DotNet.Framework.DeveloperPack_4
+Installers:
+- Architecture: x86
+  NestedInstallerType: exe
+  NestedInstallerFiles:
+  - RelativeFilePath: vistrax_setup_3.3.23061.1234.exe
+  InstallerUrl: https://update.vistrax.de/fileadmin/downloads/vistrax_setup_3.3.23061.1234.zip
+  InstallerSha256: 4960CD21818469D8136F789698B9B407071790B78625071B56388116E77E9BE4
```
Comment (stephengillie) [inline manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:26]:
```suggestion
  InstallerSha256: 4960CD21818469D8136F789698B9B407071790B78625071B56388116E77E9BE4
  InstallerSuccessCodes: 3010
```

Comment (FaserF) [reply manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:26]: Thanks, thought the default ones like 3010 would have been handled as successfully automatically

**Comment by stephengillie** [comment]
Automatic Validation ended with:
>  Manifest Error: Schema Error. [Schema validation failed. 
Error context: <root>[Installers][0][InstallerSuccessCodes] Description: Value type not permitted by 'type' constraint. 
Error context: <root>[Installers][0] Description: Failed to validate against schema associated with property name 'InstallerSuccessCodes'. 
Error context: <root>[Installers] Description: Failed to validate item #0 in array. 
Error context: <root> Description: Failed to validate against schema associated with property name 'Installers'. ] 
File: CONZEInformatik.Vistrax.installer.yaml Manifest Error: Failed to process field. [InstallerSuccessCodes]

File: manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  InstallerSuccessCodes: 3010
```
Comment (stephengillie) [inline manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:27]:
```suggestion
  InstallerSuccessCodes: '3010'
```
From the `type` error, I'm guessing this has to be a string and not an integer.

Comment (FaserF) [reply manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:27]: Thank you!

**Comment by FaserF** [comment]
I cannot see whats the issue here. 
Looks right to me when I compare it to: https://github.com/microsoft/winget-cli/blob/b96598a6038ac0fb0f58cf935eb466c4484dea3c/schemas/JSON/manifests/v1.5.0/manifest.installer.1.5.0.json#L628

**Comment by stephengillie** [comment]
>  Manifest Error: Schema Error. [Schema validation failed. 
Error context: <root>[Installers][0][InstallerSuccessCodes] Description: Value type not permitted by 'type' constraint. 
Error context: <root>[Installers][0] Description: Failed to validate against schema associated with property name 'InstallerSuccessCodes'. 
Error context: <root>[Installers] Description: Failed to validate item #0 in array. 
Error context: <root> Description: Failed to validate against schema associated with property name 'Installers'. ] File: CONZEInformatik.Vistrax.installer.yaml 
Manifest Error: Failed to process field. [InstallerSuccessCodes]

File: manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  PackageDependencies:
+    - PackageIdentifier: Microsoft.DotNet.Framework.DeveloperPack_4
+Installers:
+- Architecture: x86
+  NestedInstallerType: exe
+  NestedInstallerFiles:
+  - RelativeFilePath: vistrax_setup_3.3.23061.1234.exe
+  InstallerUrl: https://update.vistrax.de/fileadmin/downloads/vistrax_setup_3.3.23061.1234.zip
+  InstallerSha256: 4960CD21818469D8136F789698B9B407071790B78625071B56388116E77E9BE4
+  InstallerSuccessCodes: '3010'
```
Comment (stephengillie) [inline manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:27]:
```suggestion
  InstallerSuccessCodes:
  - 3010
```
@FaserF Sorry for the multiple changes. I've copied this from another PR that was accepted, and updated the integer. Hopefully this time will work.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 2023-11-07 11:29:11.092 [CORE] WinINet downloading from url: https://cdn.winget.microsoft.com/cache/manifests/m/Microsoft/DotNet/Framework/DeveloperPack_4/4.8.1/2ac0  
> 2023-11-07 11:29:11.114 [CORE] Download hash: 2c071c5b271658b6de1f2a3e8376975ae6cd9aa37f7587e5bdc52bd25ad02795  
> 2023-11-07 11:29:11.115 [CORE] Download completed.  
> 2023-11-07 11:29:11.117 [CLI ] Starting installer selection.  
> 2023-11-07 11:29:11.117 [CLI ] Installer [X64,exe,Unknown,en-US] not applicable: Architecture was excluded by caller : X64  
> 2023-11-07 11:29:11.119 [CLI ] No suitable installer found for manifest Microsoft.DotNet.Framework.DeveloperPack_4 with version 4.8.1  
> 2023-11-07 11:29:11.120 [CLI ] Terminating context: 0x8a150104 at D:\a\_work\1\s\external\pkg\src\AppInstallerCLICore\Workflows\DependenciesFlow.cpp:11a  

This situation is probably unique to the Manual Validation pipeline.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 2:27:20 PM: 2023-11-07 14:27:20.383 [CLI ] ShellExecute installer failed: 1603 
...
2:27:21 PM: Product: vistrax -- Error 4180. There was an error during the Windows features configuration process. Your original configuration will be restored. 

`1603` often means that there is a missing dependency. The "Windows features" part sometimes indicates the .NET 3.5 Framework.

File: manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+- interactive
+- silent
+InstallerSwitches:
+  Silent: /quiet
+  SilentWithProgress: /passive
+UpgradeBehavior: install
+ReleaseDate: 2023-03-02
+Dependencies:
+  PackageDependencies:
+    - PackageIdentifier: Microsoft.DotNet.Framework.DeveloperPack_4
```
Comment (stephengillie) [inline manifests/c/CONZEInformatik/Vistrax/3.3.23061.1234/CONZEInformatik.Vistrax.installer.yaml:19]:
```suggestion
    - PackageIdentifier: Microsoft.DotNet.Framework.DeveloperPack_4
    - PackageIdentifier: Microsoft.DotNet.Runtime.3_1
```

**Comment by stephengillie** [comment]
image

**Comment by stephengillie** [comment]
Running manually gives the same error, with no hints to the specific feature.
image

**Comment by FaserF** [comment]
Added two other missing preriquisits mentioned here: 
https://help.vistrax.com/de/v2/installationshilfe/voraussetzungen

**Comment by stephengillie** [comment]
This passed the Automatic Validation pipeline, but in a Windows VM a dependency had this pop-up:
image

Clicking "No" caused the install to fail. 
- Clicking "Yes" would cause a reboot. Unsure if there's a way to install but not reboot. 
- The dependency should be investigated and maybe modified to remove the prompt. 
- Should this package be approved despite the dependency prompt?

**Comment by FaserF** [comment]
I would approve the package in that state. The dependency promt seems to be an issue then with the Microsoft.DirectX winget package then, I will try to have a look into that separated

**Comment by stephengillie** [comment]
For some reason, this still fails with `ShellExecute installer failed: 3010` despite this being marked as a success code.

**Comment by FaserF** [comment]
Since I have no clue how to fix this, and it looks like a bug within winget itself, I have raised a ticket there https://github.com/microsoft/winget-cli/issues/3911

**Comment by FaserF** [comment]
I still dont think its an issue with the package itself and most likely is a issue on winget: https://github.com/microsoft/winget-cli/issues/3911

**Comment by stephengillie** [comment]
Automatic Validation ended with:
>  Installation failed with exit code -1978334972 MSI (s) (D8:E0) [14:18:29:866]: Windows Installer installed the product. Product Name: Microsoft Visual C++ 2022 X86 Minimum Runtime - 14.38.33130. Product Version: 14.38.33130. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (D8:C4) [14:18:30:991]: Windows Installer installed the product. Product Name: Microsoft Visual C++ 2022 X86 Additional Runtime - 14.38.33130. Product Version: 14.38.33130. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. [1804:0FA0][2024-01-05T14:18:27]w363: Could not create system restore point, error: 0x80070422. Continuing...

(Automated response - build 530.)

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 2024-01-08 13:25:16.493 [CLI ] Starting: 'C:\Users\User\AppData\Local\Temp\WinGet\Microsoft.DotNet.Framework.DeveloperPack_4.4.8.1\ndp481-devpack-enu.exe' with arguments '/install /passive /norestart' 2024-01-08 13:27:52.276 [CLI ] ShellExecute installer failed: 3010

(Automated response - build 530.)

**Comment by stephengillie** [comment]
There seems to be an issue with the dependency `Microsoft.DotNet.Framework.DeveloperPack_4`.

**Comment by stephengillie** [comment]
> There seems to be an issue with the dependency `Microsoft.DotNet.Framework.DeveloperPack_4`.
Apparently it requires a reboot now. 
image

**Pull request closed without merging**